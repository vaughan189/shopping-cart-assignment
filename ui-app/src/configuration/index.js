import { loginConfiguration } from "./Login/index";

export { loginConfiguration };
