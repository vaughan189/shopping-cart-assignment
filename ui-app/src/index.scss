@import url(https://fonts.googleapis.com/css?family=Dosis:400,600,700);

body {
  font-family: "Dosis", sans-serif;
  margin: 0;
}
