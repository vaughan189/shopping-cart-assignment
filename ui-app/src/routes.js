import React from "react";
import { Route, Switch } from "react-router-dom";

import Home from "./pages/Home";
import Cart from "./pages/Cart";
import NotFound from "./pages/NotFound";
import { Products } from "./pages/Products";


export const Routes = () => {
  return (
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/home" component={Home} />
      <Route exact path="/cart" component={Cart} />
      <Route exact path="/Products" component={Products} />
      <Route path="*">
      <NotFound />
    </Route>
    </Switch>
  );
};
