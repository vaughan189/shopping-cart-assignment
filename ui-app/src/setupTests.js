import { configure } from "enzyme";
import enableHooks from "jest-react-hooks-shallow";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

configure({ adapter: new Adapter() });
enableHooks(jest, { dontMockByDefault: true });
