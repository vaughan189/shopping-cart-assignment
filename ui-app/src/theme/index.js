import { createTheme } from "@mui/material";
import { green, pink } from "@mui/material/colors";

export const theme = createTheme({
  palette: {
    primary: {
      main: pink[500],
    },
    secondary: {
      main: green[500],
    },
  },
});
