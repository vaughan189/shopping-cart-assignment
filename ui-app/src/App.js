import React from "react";

import { BrowserRouter as Router } from "react-router-dom";
// import { Footer } from "./components/Footer";
import { Header } from "./components/Header";

import { Routes } from "./routes";

function App() {
  return (
    <Router>
      <React.StrictMode>
        <Header />
        <div>
          <Routes />
        </div>
        {/* <Footer /> */}
      </React.StrictMode>
    </Router>
  );
}

export default App;
