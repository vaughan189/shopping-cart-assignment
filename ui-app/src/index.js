import React from "react";
import ReactDOM from "react-dom";
import "./index.scss";
import App from "./App";
import reportWebVitals from "./reportWebVitals";
import { ThemeProvider } from "@mui/material";

import "./assets/fonts/Dosis-VariableFont_wght.ttf";

import { theme } from "./theme";

ReactDOM.render(
  <ThemeProvider theme={theme}>
    <App />
  </ThemeProvider>,
  document.getElementById("root")
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
