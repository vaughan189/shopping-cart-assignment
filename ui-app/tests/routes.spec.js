import React from "react";
import { shallow } from "enzyme";
import { Routes } from "../src/routes";

describe("Router component", () => {
  it("should render router component", () => {
    const component = shallow(<Routes />);
    expect(component).toMatchSnapshot();
  });
});
