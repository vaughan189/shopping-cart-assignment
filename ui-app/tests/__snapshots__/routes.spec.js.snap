// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router component should render router component 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/home"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/cart"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/products"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/login"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/register"
  />
  <Route
    path="*"
  >
    <NotFound />
  </Route>
</Switch>
`;
